client/src/test/java/io/pravega/client/admin/impl/StreamManagerImplTest.java

```diff
@@ -865,4 +870,93 @@ public void testListCompletedTransactions() throws ConnectionFailedException {
         assertEquals(2, listTransactions.size());
         AssertExtensions.assertThrows(Exception.class, () -> streamManager.listCompletedTransactions(new StreamImpl(scope, stream)));
     }
+
+    @Test
+    public void testGetDistanceBetweenTwoStreamCuts() {
+        String scope = "scope";
+        String stream = "stream";
+        Stream stream1 = new StreamImpl(scope, stream);
+        final StreamCut startStreamCut = getStreamCut("scope", stream, 10L, 1, 2);
+        final StreamCut endStreamCut = getStreamCut("scope", stream, 30L, 1, 2);
+
+        Controller mockController = mock(Controller.class);
+        ConnectionPoolImpl pool = new ConnectionPoolImpl(ClientConfig.builder().maxConnectionsPerSegmentStore(1).build(), connectionFactory);
+        @Cleanup final StreamManager streamManagerImpl = new StreamManagerImpl(mockController, pool);
+
+        ImmutableSet<Segment> r = ImmutableSet.<Segment>builder()
+                .addAll(startStreamCut.asImpl().getPositions().keySet())
+                .addAll(endStreamCut.asImpl().getPositions().keySet()).build();
+
+        when(mockController.getSegments(startStreamCut,
+                endStreamCut)).thenReturn(CompletableFuture.completedFuture(new StreamSegmentSuccessors(r, "")));
+        CompletableFuture<Long> cf = streamManagerImpl.getDistanceBetweenTwoStreamCuts(stream1, startStreamCut,
+                endStreamCut);
+        try {
+            Long distance = cf.join();
+            assertEquals(Long.valueOf(40), distance);
```

Comment:
Shouldn't this be 20?

Reply:
As discussed last, since we are using two offsets so it will be 40.